aries-site/src/examples/components/namevaluelist/Examples/NameValueListEditHorizontalExample.js

```diff
@@ -0,0 +1,146 @@
+import React, { Fragment, useContext, useState } from 'react';
+import { ThemeContext } from 'styled-components';
+import {
+  Anchor,
+  Box,
+  Button,
+  Form,
+  Heading,
+  MaskedInput,
+  NameValueList,
+  NameValuePair,
+  TextInput,
+  Text,
+  ResponsiveContext,
+} from 'grommet';
+import {
+  FormGrid,
+  NameValueListFormField,
+  NameValueListFormLabel,
+} from './components';
+import { companyInfomation } from '../data';
+
+export const NameValueListEditHorizontalExample = () => {
+  const theme = useContext(ThemeContext);
+  const size = useContext(ResponsiveContext);
+  const [currentData, setCurrentData] = useState(companyInfomation);
+  const [tempData, setTempData] = useState(currentData);
+  const [edit, setEdit] = useState(false);
+
+  const Wrapper = size !== 'small' ? Fragment : Box;
+
+  return (
+    <Box gap="medium">
+      <Box justify="between" width="large" direction="row" align="start">
+        <Box gap="xsmall">
+          <Heading level={2} size="small" margin="none">
+            Company Information
+          </Heading>
+          <Text size="large">Add your company information.</Text>
+        </Box>
+        {!edit && (
+          <Button secondary label="Edit" onClick={() => setEdit(true)} />
+        )}
+      </Box>
+      {!edit ? (
+        <NameValueList>
+          {Object.entries(currentData).map(([name, value]) => (
+            <NameValuePair key={name} name={<Text weight="bold">{name}</Text>}>
+              {name === 'Email' ? (
+                <Anchor label={value.value} href={`mailto:${value.value}`} />
+              ) : (
+                <Text {...theme.global.input.font}>{value.value || '--'}</Text>
+              )}
+            </NameValuePair>
+          ))}
+        </NameValueList>
+      ) : (
+        <Form
+          messages={{ required: 'This is a required field.' }}
+          onSubmit={() => {
+            setCurrentData(tempData);
+            setEdit(false);
+          }}
+        >
+          <FormGrid>
+            {Object.entries(currentData).map(([name]) => (
```

Comment:
I'd second this and want to reinforce that we'd probably never have displayable field names in data. Quake has something called 'Metabuilder' that they are trying to get away from since it obfuscates the UI code and there's always exceptions and presentation details you have to separate from the data. It's much easier from a maintenance  perspective to simply have a bunch of verbose react jsx. Most apps really will do something more like:
```jsx
<NameValueList>
  <NameValuePair name="Company" value={data.company}/>
  <NameValuePair name="Copyright" value={data.copyright}/>
  ...
</NameValueList>
```

Reply:
Thanks for this feedback. I've made adjustments to allow for render functions with data that has been formatted. For now, I've left our code as mapping through the data, but a team could implement more hard-coded like Mike has here if desired.